### DISCUSSION TITLE
FCMultiLoopTID does not expand "all" the scalar product

### DISCUSSION THREAD

**Discussion by mescia** [FeynCalc/feyncalc#112] [General]
Dear all

thanks for the work in FeynCalc and sorry to disturb you again. 

FCMultiLoopTID  perfectly works in this example FVD[q1,mu]  FAD[q1, q2, {q1 - p}, {q2 - p}, {q1 - q2}]. 
But it gets stuck if I substitute in the previous example  FVD[q1,mu] with SPD[q1,k1], namely if I want to calculate expression like this  SPD[q1,k1] FAD[q1, q2, {q1 - p}, {q2 - p}, {q1 - q2}]. As you know it is exactly the same example but now I have contracted the vector FVD[q1,mu] with a soft momentum k1.

Let me add, FCMultiLoopTID  perfectly expand the scalar product  in the case of SPD[q1,p]  FAD[q1, q2, {q1 - p}, {q2 - p}, {q1 - q2}], namely if p is also present in the denominator.

Unfortunately, in my calculation I have a soft moment k1 (from a operator with derivative). Then I expand k1 in the loop calculation. Then k1 appears in the numerator as scalar product but not in the denominator.
 
thanks and sorry. Federico.

**Comment by vsht** [comment]
Hi,

by default `FCMultiLoopTID` does not attempt any decompositions unless there are loop momenta
with uncontracted indices in the numerator or loop momenta contracted to polarization vectors and/or
Dirac matrices. This is due to the appearance of irreducible numerators beyond 1-loop, so that unit
numerators can be achieved only via IBPs but not via tensor reduction.

But there is also an option Uncontract to handle particular cases "by hand" if the user believes that
it is useful there
```
FCMultiLoopTID[ SPD[q1, k1] FAD[q1, q2, {q1 - p}, {q2 - p}, {q1 - q2}], {q1, q2},  Uncontract -> {k1}]
```

Cheers,
Vladyslav
